environments/gcp/ingress.tf

```diff
@@ -0,0 +1,77 @@
+module "nginx-controller" {
+  source  = "terraform-iaac/nginx-controller/helm"
+  version = "2.3.0"
+  namespace = "kube-system"
+depends_on = [ module.gke ]
+}
+
+data "http" "manifestfile" {
+  url = "https://github.com/cert-manager/cert-manager/releases/download/v1.13.2/cert-manager.crds.yaml"
+}
+
+data kubectl_file_documents "cert-manager-manifest" {
+  content = data.http.manifestfile.response_body 
+
+}
+
+#Installing cert-manager CRDs from the manifest instead of using Helm is the recommended approach for production environments. 
```

Comment:
Do I remember correctly that this is to avoid a chicken-egg problem with `terraform apply`?

Reply:
Partially yes, the cert-manager CRDs need to be present beforehand, and creating them through a manifest is the recommended approach as opposed to using Helm, which has a number of limitations (https://cert-manager.io/docs/installation/helm/#crd-considerations)